EIPS/eip-1775.md

```diff
@@ -0,0 +1,571 @@
+---
+eip: 1775
+title: App Keys, application specific wallet accounts
+author: Vincent Eli (@Bunjin), Dan Finlay (@DanFinlay)
+discussions-to: https://ethereum-magicians.org/t/eip-erc-app-keys-application-specific-wallet-accounts/2742
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-02-20
+requires: 137
+replaces: 1581
```

Comment:
Are you in contact with @bitgamma the author of #1581 ? It would be nice to see you come together on the best solution instead of two competing standards. Would he be willing to move his Draft to Superseded?

Reply:
Yes we are currently discussing on Eth magicians with @gravityblast on how #1581 and #1775 are related and which standard we should pick for cross applications keys (such as chat keys).

Reply:
the way #1775 is defined right now does not replace #1581 since the use-case is different. Actually #1775 could be seen as a specialization of #1581, since #1581 basically defines a fixed-prefix + a key usage identifier + an application specific part. However, unless we can agree on this common prefix (we use the m/43'/60'/1581' as suggested by @Arachnid) I'd say the two EIP cover similar but distinct areas.

Reply:
thanks @bitgamma for commenting here and on the Ethmagicians post!

One option is indeed to have the two standard coexisting: 

EIP1581 for cross application keys (non crypto accounts)
EIP1775 for application keys.
I don't think EIP1775 could be a specialization of 1581 since you propose the following full path:
m / purpose' / coin_type' / subpurpose' / key_type' / key_index
and this does not allow to have apps UIDs in the way they are defined in EIP1775 nor personas, nor app specific hd subpath. 

However as I suggested in the above discussion, EIP1775 could be used to derive cross app keys (called non-wallet accounts in EIP1581) , if we'd like to define one standard for these 2 use cases instead of 2. Then cross application keys (non wallet accounts as they are called in EIP1581) would be a subcase of EIP1775 with names used to derive UIDs not following ENS nor DNS but following a standard such as `cak:name_of_use_case`

Reply:
well the key_index in EIP1581 is actually defined as being any identifier with the only hard prescription that all components be hardened except for the last one. So the key_index could be the root if EIP-1775 (with key type being 1775'). I can also update the specs to make it clearer that the key_index is basically free-form.

Your suggestion is of using cak:name_of_use_case is interesting and would fit the use case from a functional point of view, but is a performance issue for hardware wallets, especially for Keycard which is a SmartCard (so very resource limited). Since hashes are quite long, derivation would take a lot and since every hash is totally unrelated to the other there isn't a long prefix I could cache and derive just the last 1 or 2 components.

Reply:
@bitgamma To allow better investigation of this topic, do you have some benchmark of the Keycard wrt HD path depth?

Reply:
It takes around 230-250ms per level for the Keycard. Currently 10 level is the maximum allowed (which apparently is also true for the current version of the Ledger Ethereum app, looking at its code)

Reply:
For the sake of pragmatism and flexibility, I'm removing the ` superseded / replaces ` field and treating this EIP as having a stricly different aim than what bitgamma calls "non- wallet keys".